src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -86,6 +86,9 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
     HWIntrinsicCategory    category    = HWIntrinsicInfo::lookupCategory(intrinsicId);
     size_t                 numArgs     = node->GetOperandCount();
 
+    // We need to validate that other phases of the compiler haven't introduced unsupported intrinsics
+    assert(compiler->compIsaSupportedDebugOnly(isa));
```

Comment:
This shows the assert triggering on the original bug: https://dev.azure.com/dnceng/public/_build/results?buildId=1566647&view=ms.vss-test-web.build-test-results-tab&runId=44059054&resultId=102987&paneView=debug

It's not "perfect" and its still possible for something else in the JIT to incorrectly generate an AVX2 instruction; but this should catch the bulk of them, particularly since most intrinsic usages are now driven through the HWIntrinsic nodes so that optimizations and other transforms are consistently applied.